### PR TITLE
Fixing #871, workaround of php bug [72338](https://bugs.php.net/bug.php?id=72338)

### PR DESCRIPTION
**Pull request by Ovsyanka**

### CODE REVIEW

**Comment by lucasmichot** [comment]
<?
Let me some time to review this @Ovsyanka

File: src/Carbon/Carbon.php [hunk 1]
```diff
@@ -943,7 +943,11 @@ public function tz($value)
      */
     public function setTimezone($value)
     {
-        return parent::setTimezone(static::safeCreateDateTimeZone($value));
+        $date = parent::setTimezone(static::safeCreateDateTimeZone($value));
```
Comment (lucasmichot) [inline src/Carbon/Carbon.php]: Can we use `$instance` instead?

File: tests/Carbon/PhpBug72338Test.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ *
+ * * $date->setTimezone($tz) settle this changed timestamp, even in case if $tz is not in 'HH:MM' format
+ *
+ * * $date->format('U') returns changed timestamp
+ *
+ * @link https://bugs.php.net/bug.php?id=72338 This bug on bugs.php.net
+ *
+ * @internal I use days changing in tests because using seconds|minute|hours may run setTimezone within.
+ */
+class PhpBug72338Test extends AbstractTestCase
```
Comment (lucasmichot) [inline tests/Carbon/PhpBug72338Test.php]: I think I would put this specific tests in the dedicated existing test classes

File: tests/Carbon/PhpBug72338Test.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    /**
+     * Ensures that modify dont use changed timestamp
+     */
+    public function testModify()
+    {
+        $date = Carbon::createFromTimestamp(0);
+        $date->setTimezone('+02:00');
+        $date->modify('+1 day');
+        $this->assertSame('86400', $date->format('U'));
```
Comment (lucasmichot) [inline tests/Carbon/PhpBug72338Test.php]: Maybe a stricter assertion would be fine too? as `$date->format('U')` always returns an integer.

Comment (Ovsyanka) [inline tests/Carbon/PhpBug72338Test.php]: I dont get your point. What do you mean? $date->format('U') always returns string as i know.

File: tests/Carbon/PhpBug72338Test.php (hunk 2 above)
Comment (Ovsyanka) [inline tests/Carbon/PhpBug72338Test.php]: Do you mean that you will do it by yourself? I dont know how modifying pull-request works, so if you need anything from me - just ask.

File: src/Carbon/Carbon.php (hunk 1 above)
Comment (Ovsyanka) [inline src/Carbon/Carbon.php]: Ofc, we can. Or even better, we can use $this. I will try it.

Comment (Ovsyanka) [inline src/Carbon/Carbon.php]: Okay, i made it and updated the branch

File: tests/Carbon/PhpBug72338Test.php (hunk 3 above)
Comment (Ovsyanka) [inline tests/Carbon/PhpBug72338Test.php]: If you want we can discuss all this things on Discord (it have web-interface, so you dont need ti install anything): https://discord.gg/FDbay

**Pull request merged**